File: Python/ProjectEuler0007.py

# projecteuler.net/problem=7

# Answer: 104743

# github.com/firatksee


def isPrime(number):
    if number < 2:
        return False
    if number == 2:
        return True
    if number % 2 == 0:
        return False

    for i in range(3, int(number ** 0.5) + 1, 2):
        if number % i == 0:
            return False
    return True


def nthPrime(n):
    count, number = 0, 0
    while count < n:
        number += 1
        if isPrime(number):
            count += 1
    return number


if __name__ == "__main__":
    print(nthPrime(10001))

File: JavaScript/ProjectEuler0007.js
// projecteuler.net/problem=7

// Answer: 104743

// github.com/firatksee

function isPrime(number) {
    if (number < 2) return false;
    if (number === 2) return true;
    if (number % 2 === 0) return false;

    const limit = Math.floor(Math.sqrt(number));

    for (let i = 3; i <= limit; i += 2) {
        if (number % i === 0) return false;
    }
    return true;
}

function nthPrime(n) {
    let count = 0;
    let number = 0;
    while (count < n) {
        if (isPrime(++number)) count++;
    }
    return number;
}

console.log(nthPrime(10001));
